FreeRTOS/FreeRTOS-Kernel
# Add removed unprivileged access check to ARM_CRx_MPU xPortIsAuthorizedToAccessBuffer()

Description
-----------
During PR review of the ARM_CRx_MPU port a check for permissions inside of `xPortIsAuthorizedToAccessBuffer()` was modified and introduced a new bug. The check for if the task had write permissions to a variable it was attempting to read from was removed. This PR fixes this by adding this check back in.

Test Steps
-----------

Run the following code from an unprivileged task.
```
xReturned = xTaskNotifyGive( xTaskGetCurrentTaskHandle() );
prvNotifyCheck( xReturned );
```
This will cause a failure in the existing version of xPortIsAuthorizedToAccessBuffer(). Currently there is no check for if the buffer being used is allowed write access.

This PR fixes this issue.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
